Home

PolarDL opened the pull request:
Не могу разобраться с NPE.
Запускаю приложение, появляется главная форма. В ней жму "Create Order".
В появившейся форме создания нового заказа жму "Add Items to this Order".
В появившейся форме добавления товаров к заказу указываю существующий АйДи товара (например, 32516028337722) и его количество (например, 3).
При нажатии кнопки "Add Item" в карту должна добавиться Entry. Ключ и значение есть.
Добавил перед строкой добавления в карту отладочную информацию. Приложение выводит в консоль.
Но если ключ печатается вообще без проблем, то вот значение разбавлено ошибкой (как бы перемешано):


`32516028337722     //вот отладочная инфа - toString для ключа (АйДи)
OrderPosition{        //вот это начало отладочной инфы для Значения, в следующей строке ошибка пошла
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at ru.polardl.homeshopping.GUI.AddItemsToNewOrderForm.addItemBtnActionPerformed(AddItemsToNewOrderForm.java:217)
	at ru.polardl.homeshopping.GUI.AddItemsToNewOrderForm.access$000(AddItemsToNewOrderForm.java:13)
	at ru.polardl.homeshopping.GUI.AddItemsToNewOrderForm$2.actionPerformed(AddItemsToNewOrderForm.java:76)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
       orderPosItem=Item{id=32516028337722, itemName='Стул детский складной', color='белый', price=3000.0, leftover=29}, orderPosItemQuantity=3, orderPosItemFixedPrice=3000.0, orderPosTotalPrice=9000.0}    //а вот продолжение toString для Значения
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)`



В общем, не могли бы вы запустить приложение, повторить данный данный алгоритм до появления ошибки и подсказать, в чем косяк?

PolarDL committed:
Всё приложение, за исключением GUI.

KolosovKD commented on src/ru/polardl/homeshopping/GUI/AddItemsToNewOrderForm.java:
```diff
@@ -0,0 +1,268 @@
+
+public class AddItemsToNewOrderForm extends javax.swing.JFrame {
+    
+    static HashMap<Long, OrderPosition> orderPositionMap;
```
Вы объявили переменную, но забыли создать сам объект - переменная не инициализирована.
`static HashMap<Long, OrderPosition> orderPositionMap = new HashMap();`

PolarDL committed:
Готовое приложение с интерфейсом. Переиграл логику добавления и выбора товаров и заказов для дальнейшей работы (не вписывать с клавы, 
а выбирать из списка). Ну и в целом. До идеала, конечно, далеко, но, 
более менее работает.
